nbs/07_match_collection_class.ipynb

```diff
@@ -0,0 +1,142 @@
+{
```

Comment:
<p>This feels like too little logic to warrant it's own nb, and even to warrant it's own class.</p><p> </p><p>all we need to do when matching a key is to:</p><ul><li>split it if its and edge, and then reduce to matching nodes</li><li>if its a node, check if its in mapping or collection_mapping and based on that, return the value</li></ul>
<br/>

 _Reply via <a href='https://app.reviewnb.com/DeanLight/graph_rewrite/pull/17/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Reply:
<p>We'd like to discuss about this in our meeting:</p><p>We think that this is the only way to return a list of all possible nodes values for an attribute when using this syntax: match['collection_name']['attribute'].</p><p>This is basically all the added value we have found for this class, and without this class, the user would have to iterate over the nodes and edges (which is also a fine design choice in our opinion).</p>